File: algo-daily/coding-solution/majority_element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int m_ele = 0;
        int count = 0;
        for(auto x : nums){
            
            if(count == 0){
                m_ele = x;
            }
            
            count += (x==m_ele) ? 1 : -1;
        }
        return m_ele;
        
    }
};

File: python-daily/majority_element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        m_e = None
        for ele in nums:
            if count == 0:
                m_e = ele
                
            if m_e == ele:
                count +=1
            else:
                count -=1
                
        return m_e